plugins/default/lib/disk_usage.py

```diff
@@ -1,35 +1,97 @@
 #!/usr/bin/env python
 import re
 import sys
+import json
+import argparse
 from subprocess import CalledProcessError
+from collections import Counter
 
-from openshift import oc
+import openshift
 import nagios
 
 
-check_disk_cmd = ("df", "--output=target,pcent,ipcent")
+check_disk_cmd = ("df", "--output=pcent,ipcent,target")
 # Example output:
 # /etc/hosts      19%    8%
-check_disk_output_pattern = re.compile(r"([\w/]+)\s+(\d+)%\s+(\d+)%")
+check_disk_output_pattern = re.compile(
+    r"\s*(\d+)%\s+(\d+)%\s+(.+)$")
 
 
-def parse_df(line):
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-w", "--warn", type=int, action="store",
+                        required=True, help="Warning Threshold")
+    parser.add_argument('-c', '--crit', type=int, action="store",
+                        required=True, help="Critical Threshold")
+    return parser
+
+
+def parse_df_line(line):
     mo = check_disk_output_pattern.match(line)
     if mo is None:
         return ()
-    return (mo.group(1), int(mo.group(2)), int(mo.group(3)))
+    return (mo.group(3), int(mo.group(1)), int(mo.group(2)))
+
+
+def parse_df_lines(lines):
+    return (filter(None, map(parse_df_line, lines.splitlines())))
+
+
+def analize(pod, disks, warning_threshold, critical_threshold):
+    results = []
+    for mount, space_usage, inode_usage in disks:
+        max_pcent = max(space_usage, inode_usage)
+
+        disk_status = nagios.UNKNOWN
+        if max_pcent >= critical_threshold:
+            disk_status = nagios.CRIT
+        elif max_pcent >= warning_threshold:
+            disk_status = nagios.WARN
+        elif max_pcent < warning_threshold:
+            disk_status = nagios.OK
+
+        results.append([pod, mount, space_usage, inode_usage, disk_status])
+    return results
+
 
-def analize(pcent, ipcent, warning_threshold, critical_threshold):
-    max_pcent = max(pcent, ipcent)
-    return max_pcent >= warning_threshold, max_pcent >= critical_threshold
+def report(results):
+    counts = Counter(d[4] for d in results)
```

Comment:
The cleaner way is to give names in the code, rather than comments in the PR :)

When I'm reading, the first thing that makes understanding harder is to imagine what is `d`. A "result" shortened as "r" is probably more reasonable than "d", what does "d" stand for?!

I see `d` is also used for the return value of `analize`.

What is `d[4]`? I have to go find where `results` was defined to figure out what it is... I have to search everywhere where `report` is being called. Now that I found it, I see it is constructed from calling `analize`, so I have to go see what it returns...

And the thing with index 4 is called `disk_status`.

Suggestion:

```python
unique_statuses = Counter(disk_status for pod, mount, space_usage, inode_usage, disk_status in results)
```